Support new readxl features and help links

javierluraschi opened the pull request:
This PR adds support for `n_max` and `range` fields available now in `readxl 1.0`, disabled unsupported fields for older version of `readxl `. Refactored help links from `Wizard` to `ModalBase` and added links as follows:
- Reading Excel files using readxl -> http://readxl.tidyverse.org/
- Reading rectangular data using readr -> http://readr.tidyverse.org/
- Reading statistical data using haven -> http://haven.tidyverse.org/

@jennybc Feel free to post feedback on this PR or after trying this out in tomorrows build, thanks!

<img width="1088" alt="screen shot 2017-05-02 at 1 42 38 pm" src="https://cloud.githubusercontent.com/assets/3478847/25638439/461c514c-2f3d-11e7-9d81-b3f13457c7ec.png">

`n_max` and `skip` get disabled when `range` is specified.

<img width="1084" alt="screen shot 2017-05-02 at 1 27 45 pm" src="https://cloud.githubusercontent.com/assets/3478847/25638386/1947eaf0-2f3d-11e7-89db-b21fe154e07b.png">

`n_max` and `range` are disabled when `readxl < 1.0` is installed.

<img width="1079" alt="screen shot 2017-05-02 at 1 47 02 pm" src="https://cloud.githubusercontent.com/assets/3478847/25638626/de602aa0-2f3d-11e7-842c-cb0f0488c4c9.png">

kevinushey commented on src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImportOptionsUiSav.java:
```diff
@@ -119,6 +120,16 @@ public void setImportLocation(String importLocation)
+   public HelpLink getHelpLink()
+   {
+      return new HelpLink(
+         "Reading statistical data using haven",
```
I'm sort of iffy about calling this 'statistical data' (since it somewhat implies the data you can't read with haven is not statistical).

What about something like:

```
Reading data using haven
Reading foreign datasets using haven
```

or otherwise?

javierluraschi replied:
Yes, makes sense! Changed.

jennybc commented:
Very minor point: I think `readr` is used a couple times in PR description when `readxl` is the intent.

Just based on screenshots:

It's possible all the "Import Options" should bear the exact name of the `read_excel()` argument they correspond to. The one that stands out is `Max`, which is bound to `n_max`, and should not be confused with the unexposed argument `guess_max`. And if you do it for `n_max`, maybe it's better to be consistent across the board?

I look forward to giving it a whirl!

javierluraschi commented:
@jennybc thanks you, updated PR, changed `Max` to `N Max`. Other import dialogs use the verbose names, so we would have to also change those to be consistent. We can iterate on this label if we think of a better name, ping me in Slack and I'll push the change.
